pkg/agent/config/config.go

```diff
@@ -47,13 +51,33 @@ func Get(ctx context.Context, agent cmds.Agent, proxy proxy.Proxy) *config.Node
 			case <-time.After(5 * time.Second):
 				continue
 			case <-ctx.Done():
-				logrus.Fatalf("Interrupted")
+				logrus.Fatal("Interrupted")
 			}
 		}
 		return agentConfig
 	}
 }
 
+// GetKubeProxyDisabled returns true if kube-proxy has been disabled in the server configuration.
+// The server may not have a complete view of cluster configuration until after all startup
+// hooks have completed, so a call to this will block until after the server's readyz endpoint
+// returns OK.
+func GetKubeProxyDisabled(ctx context.Context, node *config.Node, proxy proxy.Proxy) bool {
```

Comment:
Nit: the "get" is unnecessary and reads a little cleaner without it.

```suggestion
func KubeProxyDisabled(ctx context.Context, node *config.Node, proxy proxy.Proxy) bool {
```

Reply:
If I change this I'm going to change `Get(` too which also bugs me.
https://github.com/k3s-io/k3s/blob/21c8a336476638cf7cfffcfb47a31da33149ea0d/pkg/agent/config/config.go#L41

Reply:
That first "get" is reasonable for the most part. The one on 65 reads weird. "IsThingDisable() bool" vs "getIsThingDisabled() bool".